Lib/pathlib/_abc.py

```diff
@@ -16,7 +16,16 @@
 import posixpath
 from glob import _GlobberBase, _no_recurse_symlinks
 from stat import S_ISDIR, S_ISLNK, S_ISREG, S_ISSOCK, S_ISBLK, S_ISCHR, S_ISFIFO
-from ._os import UnsupportedOperation, copyfileobj
+from ._os import copyfileobj
+
+
+__all__ = ["UnsupportedOperation"]
+
+
+class UnsupportedOperation(NotImplementedError):
+    """An exception that is raised when an unsupported operation is attempted.
+    """
+    pass
```

Comment:
Cosmetically this will have tracebacks with `pathlib._abc.UnsupportedOperation`, perhaps in `pathlib.__init__` we could override the module? We do this in `abc`, ``tomllib``, perhaps elsewhere too.

``UnsupportedOperation.__module__ = __name__``

Reply:
Good shout, but I'll do this in another PR so that I can backport to 3.13.